src/lib.rs

```diff
@@ -24,3 +25,12 @@ mod theme;
 
 #[cfg(test)]
 mod test;
+
+/// Resets the cursor when a user presses `Ctrl+C`.
+/// This is useful when the cursor is hidden and the program is interrupted.
+fn show_cursor_after_ctrlc(term: &Term) {
+    let t = term.clone();
+    let _ = ctrlc::set_handler(move || {
```

Comment:
Isn't this going to trap all future ctrlc's after demand is done?

Reply:
I believe you can make use of this: https://docs.rs/signal-hook/latest/signal_hook/low_level/fn.unregister.html

It's also important this at least compiles on windows, not sure if we need the cursor logic there or not though. We should at least run ˋcargo build` in GitHub actions on this project.

Reply:
🤔 good question, guess i need to dig a bit deeper into this

Reply:
Yeah I believe that's why I didn't put it in here when I added it to mise, since in mise I can trap forever. I do think this is doable though